src/Agent.Worker/Variables.cs

```diff
@@ -589,6 +613,58 @@ public void CopyInto(Dictionary<string, VariableValue> target)
             }
         }
 
+        private Boolean IsReadOnly(Variable variable)
+        {
+            if (variable.ReadOnly)
+            {
+                return true;
+            }
+
+            if (_wellKnownSystemVariables == null)
+            {
+                _wellKnownSystemVariables = new List<String>();
+
+                // If variable is not marked as readOnly, return whether its in the set of reserved system variables (which are always readOnly)
+                List<String> readOnlyPrefixes = new List<String>()
+                {
+                    "system.",
+                    "build.",
+                    "agent."
+                };
+                // These are all classes which have system variables defined in them. See https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Microsoft.VisualStudio.Services.Agent/Constants.cs
+                List<System.Type> wellKnownSystemVariableClasses = new List<System.Type>()
```

Comment:
Are there vars in here that don't start with the mentioned perfixes?

Reply:
Oh whoops - I guess not, so I guess we don't need that check at all - good catch.

Reply:
I take that back - there are in `WellKnownDistributedTaskVariables` I believe